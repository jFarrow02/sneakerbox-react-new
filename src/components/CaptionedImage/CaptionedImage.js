import React from 'react';
import './CaptionedImage.scss';

import propTypes from 'prop-types';

const CaptionedImage = props=>{

    return(
        <div className='CaptionedImage'>
            <div 
                className='CaptionedImage--caption'
                style={{...props.captionStyles}}
            >
                {props.caption}
            </div>
            {props.children}
        </div>
    )
}

export default CaptionedImage;

CaptionedImage.propTypes = {
    caption: propTypes.oneOfType(
        [
            propTypes.element,
            propTypes.string,
        ]
    ),
    captionStyles: propTypes.object,
};

CaptionedImage.defaultProps = {
    caption: '',
    captionStyles: {},
}