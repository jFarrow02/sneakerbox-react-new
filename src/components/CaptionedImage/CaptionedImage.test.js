import React from 'react';
import renderer from 'react-test-renderer';
import CaptionedImage from './CaptionedImage';

it('should match a snapshot', ()=>{
    const tree = renderer.create(<CaptionedImage/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
})