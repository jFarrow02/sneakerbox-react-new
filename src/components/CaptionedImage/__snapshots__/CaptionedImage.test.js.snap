// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match a snapshot 1`] = `
<div
  className="CaptionedImage"
>
  <div
    className="CaptionedImage--caption"
    style={Object {}}
  >
    
  </div>
</div>
`;
